contribs/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaObservableQueue.java

```diff
@@ -0,0 +1,230 @@
+    this.longPollTimeout = config.getIntProperty("workflow.dyno.queues.longPollTimeout", 1000);
+    // Init Kafka producer and consumer properties
+    Properties producerProps = new Properties();
+    Properties consumerProps = new Properties();
+    consumerProps.put("group.id", queueName + "_group");
+    String serverId = config.getServerId();
+    consumerProps.put("client.id", queueName + "_consumer_"+serverId);
+    producerProps.put("client.id", queueName + "_producer_"+serverId);
+    Map<String, Object> configMap = config.getAll();
+    for (Entry<String, Object> entry : configMap.entrySet()) {
+      String key = entry.getKey();
+      String value = (String) entry.getValue();
+      if (key.startsWith(KAFKA_PREFIX)) {
+        if (key.startsWith(KAFKA_PRODUCER_PREFIX)) {
+          producerProps.put(key.replaceAll(KAFKA_PRODUCER_PREFIX, ""), value);
+        } else if (key.startsWith(KAFKA_CONSUMER_PREFIX)) {
+          consumerProps.put(key.replaceAll(KAFKA_CONSUMER_PREFIX, ""), value);
+        } else {
+          producerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+          consumerProps.put(key.replaceAll(KAFKA_PREFIX, ""), value);
+        }
+      }
+    }
+
+    // Init Kafka producer and consumer
+    producer = new KafkaProducer<>(producerProps);
+    consumer = new KafkaConsumer<>(consumerProps);
+    consumer.subscribe(Collections.singletonList(queueName));
+    logger.info("KafkaObservableQueue initialized for {}", queueName);
+  }
+
+  @Override
+  public Observable<Message> observe() {
+    OnSubscribe<Message> subscriber = getOnSubscribe();
+    return Observable.create(subscriber);
+  }
+
+  @Override
+  public List<String> ack(List<Message> messages) {
+    return messages.stream().map(Message::getId).collect(Collectors.toList());
+  }
+
+  public void setUnackTimeout(Message message, long unackTimeout) {
+  }
+
+  @Override
+  public void publish(List<Message> messages) {
+    publishMessages(messages);
+  }
+
+  @Override
+  public long size() {
+    return 0;
+  }
+
+  @Override
+  public String getType() {
+    return QUEUE_TYPE;
+  }
+
+  @Override
+  public String getName() {
+    return queueName;
+  }
+
+  @Override
+  public String getURI() {
+    return queueName;
+  }
+
+  private List<Message> receiveMessages() {
+    List<Message> messages = new ArrayList<>();
+    try {
+
+      ConsumerRecords<String, String> records = consumer.poll(longPollTimeout);
+      if (records.count() == 0) {
+        return messages;
+      }
+
+      logger.info("polled {} messages from kafka topic.", records.count());
+      records.forEach(record -> {
+        logger.debug("Consumer Record: key: {}, value: {}, partition: {}, offset: {}", record.key(),
+            record.value(), record.partition(), record.offset());
+        Message message =
+            new Message(record.key() +":"+ record.partition(), String.valueOf(record.value()), "");
```

Comment:
Without the use of the deserializer in the consumer does this code throw runtime exception ?

Reply:
done.